File: 3_array_problems/easy/_1_largest_element_in_arr.java
import java.util.*;

class Solution {
  public int largest(ArrayList<Integer> arr) {
    int largest = arr.get(0);
    for (int i = 0; i < arr.size(); i++) {
      if (arr.get(i) > largest) {
        largest = arr.get(i);
      }
    }
    return largest;
    // Collections.sort(arr);
    // return arr.get(arr.size() - 1);
  }
}

public class _1_largest_element_in_arr {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int t = sc.nextInt();
    sc.nextLine();
    while (t-- > 0) {
      String inputString = sc.nextLine();
      ArrayList<Integer> input = new ArrayList<>();
      Scanner ss = new Scanner(inputString);
      while (ss.hasNextInt()) {
        int num = ss.nextInt();
        input.add(num);
      }
      Solution ob = new Solution();
      System.out.println(ob.largest(input));
      System.out.println("~");
      ss.close();
    }
    sc.close();
  }
}

File: 3_array_problems/easy/1_largest_element_in_arr.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
#include <iostream>
#include <sstream>
#include <string>
#include <vector>
using namespace std;

// } Driver Code Ends

// User function Template for C++

class Solution {
public:
  // int largest(vector<int> &arr) {
  //   sort(arr.begin(), arr.end());
  //   return arr[arr.size() - 1];
  // }
  // TC: O(NlogN)
  // OC: O(1)
  int largest(vector<int> &arr) {
    // code here
    int largest = arr[0];
    for (int i = 0; i < arr.size(); i++) {
      if (arr[i] > largest) {
        largest = arr[i];
      }
    }
    return largest;
  }
  // TC: O(N)
  // OC: O(1)
};

//{ Driver Code Starts.

int main() {

  int t;
  cin >> t;
  cin.ignore();

  while (t--) {
    string input;
    int num;
    vector<int> arr;

    getline(cin, input);
    stringstream s2(input);
    while (s2 >> num) {
      arr.push_back(num);
    }

    Solution ob;
    cout << ob.largest(arr) << endl;
    cout << "~" << endl;
  }
  return 0;
}

// } Driver Code Ends
